lib/src/signed_video_authenticity.c

```diff
@@ -122,31 +86,25 @@ allocate_memory_and_copy_string(char **dst_str, const char *src_str)
  * Group of functions that performs transfer operations between structs.
  */
 
-#ifdef ACCUMULATED_VALIDATION
-static svi_rc
-transfer_accumulated_validation(signed_video_accumulated_validation_t *dst,
-    const signed_video_accumulated_validation_t *src)
+svi_rc
+transfer_product_info(signed_video_product_info_t *dst, const signed_video_product_info_t *src)
 {
-  assert(dst && src);
+  // For simplicity we allow nullptrs for both |dst| and |src|. If so, we take no action and return
+  // SVI_OK.
+  if (!src || !dst) return SVI_OK;
```

Comment:
The diff has messed up. This is actually old code. Back in the days when this was added, this function was (and may still be) a helper in other places and there was no guarantee that inputs where valid. To avoid breaking things, NULL ptr were allowed. We should revisit this and replace with an assert or follow your suggestion. But that should be in a separate PR.